sdk/src/main/scala/io/horizen/account/api/rpc/service/Backend.scala

```diff
@@ -57,49 +59,100 @@ object Backend {
       maxPrice: BigInteger = MAX_GAS_PRICE,
       ignorePrice: BigInteger = BigInteger.TWO
   ): BigInteger = {
-    val blockHeight = history.getCurrentHeight
-    // limit the range of blocks by the number of available blocks and cap at 1024
-    val blocks: Integer = (blockCount * 2).min(blockHeight).min(1024)
-
-    // define limit for included gas prices each block
-    val limit = 3
-    val prices: Seq[BigInteger] = {
-      var collected = 0
-      var moreBlocksNeeded = false
-      // Return lowest tx gas prices of each requested block, sorted in ascending order.
-      // Queries up to 2*blockCount blocks, but stops in range > blockCount if enough samples were found.
-      (0 until blocks).withFilter(_ => !moreBlocksNeeded || collected < 2).map { i =>
-        val block = history
-          .blockIdByHeight(blockHeight - i)
-          .map(ModifierId(_))
-          .flatMap(history.getStorageBlockById)
-          .get
-        val blockPrices = getBlockPrices(block, ignorePrice, limit)
-        collected += blockPrices.length
-        if (i >= blockCount) moreBlocksNeeded = true
-        blockPrices
+    var number = history.getCurrentHeight
+    val headHash = history.bestBlockId
+    val (lastHead : Option[ModifierId], lastPrice: BigInteger) = tipCache.fold(
+      (Option.empty[ModifierId], BigInteger.ZERO)
+    )(cache =>
+      (Some(cache.blockHash), cache.value)
+    )
+
+    // If the latest gasprice is still valid, return it.
+    if (lastHead.isDefined && lastHead == headHash){
```

Comment:
If I have understood correctly, `lastHead` is an Option[ModifierId] and `headHash` is a ModifierId, shouldn't the comparison always fail?